public/app/plugins/datasource/tempo/resultTransformer.test.ts

```diff
@@ -54,3 +54,70 @@ describe('transformFromOTLP()', () => {
     expect(res.data[0]).toMatchObject(otlpDataFrameFromResponse);
   });
 });
+
+describe('transformFromOTLP()', () => {
+  // Mock the console error so that running the test suite doesnt throw the error
+  const origError = console.error;
+  const consoleErrorMock = jest.fn();
+  afterEach(() => (console.error = origError));
+  beforeEach(() => (console.error = consoleErrorMock));
+
+  test('if passed bad data, will surface an error', () => {
+    const res = transformFromOTLP(
+      (badOTLPResponse.batches as unknown) as collectorTypes.opentelemetryProto.trace.v1.ResourceSpans[],
+      false
+    );
+
+    expect(res.data[0]).toBeFalsy();
+    expect(res?.error?.message).toContain('JSON is not valid OpenTelemetry format');
```

Comment:
First conditional is unnecessary as res is defined

Reply:
Also would consider switching toContain(string) to expect the message to be truthy